react/features/filmstrip/subscriber.web.js

```diff
@@ -153,3 +158,96 @@ StateListenerRegistry.register(
             store.dispatch(setTileViewDimensions(gridDimensions));
         }
     });
+
+/**
+ * Listens for changes to the remote participant count to recompute the reordered list of the remote participants.
+ */
+StateListenerRegistry.register(
+    /* selector */ getRemoteParticipantCount,
+    /* listener */ (count, store) => _updateRemoteParticipants(store));
+
+/**
+ * Listens for changes to the screensharing status of the remote participants to recompute the reordered list of the
+ * remote endpoints.
+ */
+StateListenerRegistry.register(
+    /* selector */ state => state['features/video-layout'].remoteScreenShares,
+    /* listener */ (remoteScreenShares, store) => _updateRemoteParticipants(store));
+
+StateListenerRegistry.register(
+    /* selector */ state => state['features/base/participants'].dominantSpeaker,
+    /* listener */ (dominantSpeaker, store) => _updateRemoteParticipants(store));
+
+/**
+ * Private helper to calculate the reordered list of remote participants.
+ *
+ * @param {*} store - The redux store.
+ * @returns {void}
+ */
+function _updateRemoteParticipants(store) {
+    const state = store.getState();
+    const { dominantSpeaker, local, previousSpeakers, remote } = state['features/base/participants'];
+
+    if (!remote) {
+        return;
+    }
+    const { remoteParticipants, visibleParticipants } = state['features/filmstrip'];
+    const currentParticipants = Array.from(remote.values());
+    const { remoteScreenShares } = state['features/video-layout'];
+
+    // Make a copy so that we don't modify the values in redux.
+    const screenShares = (remoteScreenShares || []).slice();
+    const compareDisplayNames = function(a, b) {
+        const nameA = a.name.toUpperCase();
+        const nameB = b.name.toUpperCase();
+        const result = nameA > nameB
+            ? 1
+            : nameA < nameB ? -1 : 0;
+
+        return result;
+    };
+
+    // Sort the remote participants alphabetically.
+    let reorderedParticipants
+        = currentParticipants
+            .sort(compareDisplayNames)
+            .map(p => p.id);
+
+    const speakersList = [];
+
+    dominantSpeaker && speakersList.push(dominantSpeaker);
+
+    // Check if the previous speakers are still in the conference. Ideally, the bridge should send an updated dominant
+    // speaker event when any of the previous speakers leave the conference.
+    // TODO: Remove this check when that is fixed in the bridge.
+    if (previousSpeakers?.length) {
+        const connected = previousSpeakers.filter(speaker => currentParticipants.findIndex(p => p === speaker) >= 0);
```

Comment:
Is `currentParticipants.findIndex(p => p === speaker) >= 0` will be more expensive than `remote.has(speaker)`?

I think it is. `findIndex` should have linear complexity (O(n)) and since remote is a Map `has` should have constant complexity (O(1)).

Reply:
Good point!

Reply:
I'm also wondering if it will be better to do the filtering before `previousSpeakers` have reached the redux store?

Reply:
That wouldn't help since we are trying to eliminate the users that have left after the last dominant speaker event was received. When the event is received, the list of previous speakers is up to date and it doesn't need to be filtered while updating the value in the redux store.

Reply:
I meant filtering them when somebody leaves instead of filtering them here. WDYT?